src/main/kotlin/com/terraformation/backend/customer/model/SimpleUserModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.terraformation.backend.customer.model
+
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.default_schema.UserId
+import com.terraformation.backend.i18n.Messages
+import com.terraformation.backend.log.perClassLogger
+
+data class SimpleUserModel(
+    val userId: UserId,
+    val fullName: String = "",
+) {
+  companion object {
+    private val log = perClassLogger()
+
+    private fun emailIsTf(email: String): Boolean =
+        email.endsWith(suffix = "@terraformation.com", ignoreCase = true) ||
+            email == SystemUser.USERNAME
+
+    fun create(
+        userId: UserId,
+        fullName: String,
+        email: String,
+        userIsInSameOrg: Boolean,
+        userIsDeleted: Boolean,
+        messages: Messages,
+    ): SimpleUserModel {
+      return when {
+        (!userIsDeleted && (userIsInSameOrg || currentUser().canReadUser(userId))) ->
+            SimpleUserModel(userId, fullName)
+        (emailIsTf(email)) -> SimpleUserModel(userId, messages.terraformationTeam())
+        (userIsDeleted) -> SimpleUserModel(userId, messages.formerUser())
+        else -> {
+          log.warn("User {} cannot read name of user {}", currentUser().userId, userId)
```

Comment:
Should this be an error so we'll get notified about it? I assume this will still be a sign of a bug if it occurs.